### PR TITLE
RMMIS-5831: Fix mask being applied to every field after input on any …

### PR DESCRIPTION
**Pull request by kjschulz**
RMMIS-5831: Fix mask being applied to every field after input on any masked field

### CODE REVIEW

File: src/Input.jsx [hunk 1]
```diff
@@ -86,8 +86,8 @@ module.exports = React.createClass({
       var maskConfig = inputMaskUtils.getMaskConfig(this.props.mask, customConfig);
       event.keyPressed = this.state.keyPressed;
       event.unmasked = this.state.unmasked;
-      event.stateChange = inputMaskUtils.getMaskedOutput(maskConfig, event);
```
Comment (mallenscdhhs) [inline src/Input.jsx]: The problem here is that we are modifying a global object, `event` which is used for all input `onChange` events...why that is, IDK. The solution here would be something like:
```javascript
let _event = Immutable.Map(event);
_event.set('stateChange', InputMaskUtils.getMaskedOutput(maskConfig, _event.toJSON());
...
this.onChange(_event.toJSON());
```

Comment (kjschulz) [inline src/Input.jsx]: Updated...get this thang up on the bamBOOMage!

File: src/Input.jsx [hunk 2]
```diff
@@ -57,41 +58,46 @@ module.exports = React.createClass({
...
-      var maskConfig = inputMaskUtils.getMaskConfig(this.props.mask, customConfig);
-      event.keyPressed = this.state.keyPressed;
-      event.unmasked = this.state.unmasked;
-      event.stateChange = inputMaskUtils.getMaskedOutput(maskConfig, event);
+      let maskConfig = inputMaskUtils.getMaskConfig(this.props.mask, customConfig);
+      let _event = Immutable.Map(event).withMutations(evt => {
+        evt
+          .set('keyPressed', this.state.keyPressed)
+          .set('unmasked', this.state.unmasked)
+          .set('stateChange', inputMaskUtils.getMaskedOutput( maskConfig, evt.toJSON() ));
```
Comment (mallenscdhhs) [inline src/Input.jsx]: remove whitespace from function call: `getMaskedOutput(maskConfig, evt.toJSON())`

File: src/Input.jsx [hunk 3]
```diff
@@ -57,41 +58,46 @@ module.exports = React.createClass({
...
+      });
+      this.onChange( _event.toJSON() );
```
Comment (mallenscdhhs) [inline src/Input.jsx]: instead of having two `this.onChange` calls, just have one that uses a scoped `_event` value.

**Comment by traviscooperscdhhs** [comment]
:+1:

File: src/Input.jsx (hunk 3 above)
Comment (kjschulz) [inline src/Input.jsx]: How about something like that? I like what you proposed but can't reassign anything back to `_event`.  This still provides scoped output for one `this.onChange()` call

File: src/Input.jsx (hunk 2 above)
Comment (kjschulz) [inline src/Input.jsx]: Updated.

**Pull request merged**